javascript/verify/src/main.js

```diff
@@ -0,0 +1,60 @@
+const Yargs = require('yargs');
+const Jws = require('jws');
+const { Base64 } = require('js-base64');
+const JwksClient = require('jwks-rsa');
+
+const getArgs = () => {
+    const requiredString = {
+        type: 'string',
+        requiresArg: true,
+        demandOption: true,
+    };
+    return Yargs.options({
+        "webhook-body": { description: 'The unmodified webhook POST body', ...requiredString },
+        "tl-signature": { description: 'The `X-TL-Signature` webhook POST header', ...requiredString },
+    }).argv;
+}
+
+const main = async () => {
+    const {
+        "webhook-body": webhookBody,
+        "tl-signature": tlSignature,
+    } = getArgs();
+
+    try {
+        await verifyTruelayerWebhook(webhookBody, tlSignature);
+        console.log("Webhook verified ✓");
+    } catch (e) {
+        console.error("Webhook verification failed:", e.message);
+    }
+};
+
+// Verifies a truelayer webhook `body` & `tlSignature`.
+//
+// Throws an exception for invalid body + signatures.
+const verifyTruelayerWebhook = async (body, tlSignature) => {
+    // Construct a full jws using the `body` & detached jws
+    const signatureParts = tlSignature.split('.');
+    const jws = `${signatureParts[0]}.${Base64.encode(body, true)}.${signatureParts[2]}`;
+
+    const { header } = Jws.decode(jws);
+    const publicKey = await fetchJwksPublicKey(header.jku, header.kid);
+
+    // Verify using the public key
+    if (!Jws.verify(jws, header.alg, publicKey))
+        throw new Error("Invalid signature");
```

Comment:
{ } man 😂

Reply:
the js parser says I'm cool to do this :smile:

Reply:
![image](https://user-images.githubusercontent.com/9030768/128329762-1b73a42b-3be6-4d38-8817-4c4cc7f1487e.png)

Reply:
I've added braces!